### PR TITLE
Support for C++11 <random>

### PR DESCRIPTION
**Pull request by kaimast**
I adopted the C++11 random header to use the SGX provided randomness internally. Might be a good thing to merge upstream.

### CODE REVIEW

File: psw/ae/aesm_service/source/core/ipc/UnixCommunicationSocket.cpp [hunk 1]
```diff
@@ -215,7 +215,7 @@ bool UnixCommunicationSocket::init()
         memset(&serv_addr, 0, sizeof(struct sockaddr_un));
         serv_addr.sun_family = AF_UNIX;
         memset(serv_addr.sun_path, 0, sizeof(serv_addr.sun_path));
-        strncpy(serv_addr.sun_path, mSocketBase, sizeof(serv_addr.sun_path));
+        strncpy(serv_addr.sun_path, mSocketBase, sizeof(serv_addr.sun_path)-1);
```
Comment (kaimast) [inline psw/ae/aesm_service/source/core/ipc/UnixCommunicationSocket.cpp]: This is not directly to the merge request but valgrind noticed this potential buffer overflow.

**Comment by ZongminGu** [comment]
Thanks for your code. I will review it.

**Comment by lzha101** [comment]
We chose a similar way to enable C++11 <random> in Linux 2.8 release. Seems this patch is not needed.  @kaimast, could you help to check and close this?

**Comment by LuminousXLB** [comment]
@lzha101 
I'm reading [_Intel® Software Guard Extensions (Intel® SGX) SDK for Linux* OS_](https://download.01.org/intel-sgx/sgx-linux/2.9.1/docs/Intel_SGX_Developer_Reference_Linux_2.9.1_Open_Source.pdf) which was published at 2020-May-21 06:51.

In the appendix called _Unsupported C++ Standard Classes and Functions_, it says that the C++11 `<random>` library is still unsupported.
image

Could you please offer a further instruction on how to utilizing your C++11 random library?

**Comment by xxu36** [comment]
@LuminousXLB The doc needs an update. The random functionality is already there in the SDK. Thanks

**Comment by kaimast** [comment]
Going to close this pull request as I guess it wont be needed anymore.

**Pull request closed without merging**